include/nbl/builtin/hlsl/bxdf/common.hlsl

```diff
@@ -426,6 +426,32 @@ struct quotient_and_pdf
 using quotient_and_pdf_scalar = quotient_and_pdf<float>;
 using quotient_and_pdf_rgb = quotient_and_pdf<float3>;
 
+
+// Utility class
+// Making it easier to conform your BxDFs to the concept
+template <class Spectrum, class Pdf, class Sample, class Interaction, class MicrofacetCache>
+struct BxDFBase
+{
+    // BxDFs must define such typenames:
+    using spectrum_t =      Spectrum;
+    using pdf_t =           Pdf;
+    using q_pdf_t =         quotient_and_pdf<spectrum_t>;
+
+    using sample_t =        Sample;
+    using interaction_t =   Interaction;
+    using cache_t =         MicrofacetCache;
```

Comment:
sample should always be the same thing, cache should tell us the interaction (iso vs aniso)
```cpp
using sample_t = LightSample<ray_distribution_t>;
using interaction_t = MicrofacetCache::interaction_t<ray_distribution_t>;
using cache_t = MicrofacetCache;
```

P.S. I'm not sure whether HLSL202x lets us define template aliases, please check add this into `AnisotropicMicrofacetCache` and see if DXC chokes on this when actually used:
```cpp
public:

template<class ray_distribution_t>
using interaction_t = surface_interactions::Anisotropic<ray_distribution_t>;
```
if it doesn't then use similar trick on Isotropic

Reply:
>  I'm not sure whether HLSL202x lets us define template aliases

i assumed it doesnt, i'll try to make sure 
its a pity hlsl2021 has no spec XD the only way to check things is to write code and see what dxc spits out

Reply:
yeah, HLSL supports templated `using` statements for typedefs/aliases